insights/core/__init__.py

```diff
@@ -1663,6 +1663,65 @@ def path_entry(self, path):
         return self.listings[directory]['entries'][name]
 
 
+class DiskUsage(CommandParser, dict):
+    """
+    Reads output of du command and turns it into a dictionary with pathname as
+    key and size in blocks.
+
+    Supports parsing input data as long as output is 2 column data with first
+    column as space size with integer values only and second as pathname which
+    can be a file or directory. Space size with decimal values or unit suffixes
+    like M, GiB is not supported.
+
+    du command produces output in 1K blocks unless block size is specified in
+    command options or an environment variable. This parser is intended to be
+    used only with default block size of 1K which is also equal to plain "du"
+    or "du -k".
+
+    Sample input data:
+
+        56      /var/lib/alternatives
+        4       /var/lib/logrotate
+        5492	/var/lib/mlocate
+        20      /var/lib/NetworkManager
+        186484  /var/lib/pgsql
+        856	    /var/lib/rhsm
+        110712	/var/lib/rpm
+        4	    /var/lib/rsyslog
+        64	    /var/lib/systemd
+        15200	/var/lib/yum
+    """
```

Comment:
@Rohan21Lobo since this is a base class that can be used in other parsers please include an `Examples:` section to demonstrate how it is used and please add a `Raises:` section to your documentation here to identify the exceptions raised.  In particular include the `ContentException` that can be raised by the `CommandParser`.

Reply:
Done